### ISSUE TITLE
`/datum/outfit/consumed_miner` is causing unit tests to fail

### ISSUE THREAD

**Issue by SabreML** [tgstation/tgstation#66313]
```
[01:20:07] Runtime in outfit_sanity.dm,35: Cannot execute null.on outfit equip().
  proc name: Run (/datum/unit_test/outfit_sanity/Run)
  src: /datum/unit_test/outfit_sanity (/datum/unit_test/outfit_sanity)
  call stack:
  /datum/unit_test/outfit_sanity (/datum/unit_test/outfit_sanity): Run()
  RunUnitTest(/datum/unit_test/outfit_sanity (/datum/unit_test/outfit_sanity), /list (/list))
  RunUnitTests()
  /datum/callback (/datum/callback): InvokeAsync()
FAIL: /datum/unit_test/outfit_sanity 8.5s
	REASON #1: Legion-Consumed Miner's r_pocket is invalid! Could not equip a /obj/item/stack/spacecash/c1000 into that slot.
	REASON #2: [01:20:07] Runtime in outfit_sanity.dm,35: Cannot execute null.on outfit equip().
  proc name: Run (/datum/unit_test/outfit_sanity/Run)
  src: /datum/unit_test/outfit_sanity (/datum/unit_test/outfit_sanity)
  call stack:
  /datum/unit_test/outfit_sanity (/datum/unit_test/outfit_sanity): Run()
  RunUnitTest(/datum/unit_test/outfit_sanity (/datum/unit_test/outfit_sanity), /list (/list))
  RunUnitTests()
  /datum/callback (/datum/callback): InvokeAsync()
```

~~This is (most likely) being caused by the 'Legion-Consumed Miner' outfit not having a uniform set, but its `pre_equip()` proc still attempting to put items into the pockets anyway.~~
Edit: This is (most likely) being caused by both stacks of spacecash being picked at the same time, and the right pocket merging into the left pocket through `/obj/item/stack/Initialize()`. (Left pocket spawns first)
Since this deletes one of the stacks, `get_item_by_slot(r_pocket)` would return `null`, failing the unit test.
https://github.com/tgstation/tgstation/blob/9c0a04fb770807f9a45fa7f482be0172f38494a8/code/modules/mob_spawn/corpses/mining_corpses.dm#L93-L96
https://github.com/tgstation/tgstation/blob/731f8f7f5b993ce46ee74ae0bfd9e5cf5ed4aef0/code/datums/outfit.dm#L223-L226
https://github.com/tgstation/tgstation/blob/731f8f7f5b993ce46ee74ae0bfd9e5cf5ed4aef0/code/game/objects/items/stacks/stack.dm#L70-L77

**Comment by LemonInTheDark** [comment]
what you are seeing is likely the 30% * 10% chance that two spacecash stacks will spawn, so if they both get picked they merge, which triggers the unit test

**Comment by SabreML** [comment]
Ah, yep that'd do it. I wasn't aware it automatically merged between pockets like that.

**Comment by LemonInTheDark** [comment]
It's really stupid, but fixing it would be suffering. See https://github.com/tgstation/tgstation/issues/60901

**Comment by TheBonded** [comment]
I have no idea what to label this as but I dont like unlabeled prs, @LemonInTheDark is this correct?

**Comment by LemonInTheDark** [comment]
👍

**Comment by dragomagol** [comment]
Fixed by #66495
